app/mailers/role_change_mailer.rb

```diff
@@ -5,12 +5,12 @@ class RoleChangeMailer < ApplicationMailer
     metadata = {}
 
     # Populate the metadata list.
-    case role.group.group_type
+    case UserRole.group(role).group_type
     when UserGroup.group_types[:delegate_regions]
-      metadata[:region_name] = role.group.name
-      metadata[:status] = I18n.t("enums.user.role_status.delegate_regions.#{role.metadata.status}")
+      metadata[:region_name] = UserRole.group(role).name
+      metadata[:status] = I18n.t("enums.user.role_status.delegate_regions.#{UserRole.status(role)}")
     when UserGroup.group_types[:translators]
-      metadata[:locale] = role.group.metadata.locale
+      metadata[:locale] = UserRole.group(role).metadata.locale
     end
```

Comment:
You should probably store `UserRole.group(role)` in its own variable, because it's used at three times here within six lines of code)

Reply:
Done